### PR TITLE
Calling /upsert-file should be setting source as type 'file'

### PR DESCRIPTION
**Pull request by p2c2e**
I believe /upsert-file should be setting the source to 'file' type. Currently it is left as null and hence results in an error during query as follows:

Error: 1 validation error for DocumentChunkWithScore
metadata -> source
  value is not a valid enumeration member; permitted: 'email', 'file', 'chat' (type=type_error.enum; enum_values=[<Source.email: 'email'>, <Source.file: 'file'>, <Source.chat: 'chat'>])

Backend datastore is redis-stack-server

### CODE REVIEW

**Review by CyberAmyntas — COMMENTED** [review]
If you want to set the sourcetype, please use `Source.file` from `model.models`

**Comment by CyberAmyntas** [comment]
You will also need to use a Filter and specify the source to search when querying with Redis else the same error occurs.

```
{
  "queries": [
    {
      "query": "string,
      "filter": {
        "source": "file"
      },
      "top_k": 3
    }
  ]
}
```

File: services/file.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ async def get_document_from_file(file: UploadFile) -> Document:
     print(f"extracted_text:")
     # get metadata
     metadata = DocumentMetadata()
+    metadata.source = "file"
```
Comment (CyberAmyntas) [inline services/file.py:19]: When setting source, using Source.file from model.models is required vs the string

Comment (p2c2e) [reply services/file.py]: Of course. Hardcoding was a mistake - fixed that and tested. 
Also: I don't think it is mandatory to use the Source.type in the filter during query. It was tested and works without that.

**Comment by jonghoonseo** [comment]
I think https://github.com/openai/chatgpt-retrieval-plugin/issues/37#issuecomment-1483635163 is better solution.
While I tested using qdrant sample document (`examples/docker/qdrant/documents.json`), they don't have `source` field and server occurs the same error.

**Comment by p2c2e** [comment]
> I think [#37 (comment)](https://github.com/openai/chatgpt-retrieval-plugin/issues/37#issuecomment-1483635163) is better solution. While I tested using qdrant sample document (`examples/docker/qdrant/documents.json`), they don't have `source` field and server occurs the same error.

Hi - The solution you are referring to is a workaround that needs to be done at each datastore implementation level. This one addresses this at the root cause - since this will happen for all datastore types.

**Comment by jasonhuhx** [comment]
This worked for me.

**Comment by SML0127** [comment]
I reported same [issue](https://github.com/openai/chatgpt-retrieval-plugin/issues/75), and it worked for me.

**Comment by isafulf** [comment]
Thank you all!
Resolved here: https://github.com/openai/chatgpt-retrieval-plugin/commit/e07ffedb3f0365c3839b3681cc3781174634f471

**Pull request closed without merging**